packages/block-editor/src/components/block-list/block-html.js

```diff
@@ -3,75 +3,48 @@
  * External dependencies
  */
 import TextareaAutosize from 'react-autosize-textarea';
-import { isEqual } from 'lodash';
 
 /**
  * WordPress dependencies
  */
-import { Component } from '@wordpress/element';
-import { compose } from '@wordpress/compose';
+import { useEffect, useState } from '@wordpress/element';
+import { useSelect, useDispatch } from '@wordpress/data';
 import { getBlockAttributes, getBlockContent, getBlockType, isValidBlockContent, getSaveContent } from '@wordpress/blocks';
-import { withSelect, withDispatch } from '@wordpress/data';
 
-export class BlockHTML extends Component {
-	constructor( props ) {
-		super( ...arguments );
-		this.onChange = this.onChange.bind( this );
-		this.onBlur = this.onBlur.bind( this );
-		this.state = {
-			html: props.block.isValid ? getBlockContent( props.block ) : props.block.originalContent,
-		};
-	}
-
-	componentDidUpdate( prevProps ) {
-		if ( ! isEqual( this.props.block.attributes, prevProps.block.attributes ) ) {
-			this.setState( {
-				html: getBlockContent( this.props.block ),
-			} );
-		}
-	}
-
-	onBlur() {
-		const { html } = this.state;
-		const blockType = getBlockType( this.props.block.name );
-		const attributes = getBlockAttributes( blockType, html, this.props.block.attributes );
+function BlockHTML( { clientId } ) {
+	const [ html, setHtml ] = useState( '' );
+	const { block } = useSelect( ( select ) => ( {
+		block: select( 'core/block-editor' ).getBlock( clientId ),
+	} ), [ clientId ] );
+	const { updateBlock } = useDispatch( 'core/block-editor' );
+	const onChange = () => {
+		const blockType = getBlockType( block.name );
+		const attributes = getBlockAttributes( blockType, html, block.attributes );
 
 		// If html is empty  we reset the block to the default HTML and mark it as valid to avoid triggering an error
 		const content = html ? html : getSaveContent( blockType, attributes );
 		const isValid = html ? isValidBlockContent( blockType, attributes, content ) : true;
 
-		this.props.onChange( this.props.clientId, attributes, content, isValid );
+		updateBlock( clientId, { attributes, originalContent: content, isValid } );
 
 		// Ensure the state is updated if we reset so it displays the default content
 		if ( ! html ) {
-			this.setState( { html: content } );
+			setHtml( { content } );
 		}
-	}
-
-	onChange( event ) {
-		this.setState( { html: event.target.value } );
-	}
-
-	render() {
-		const { html } = this.state;
-		return (
-			<TextareaAutosize
-				className="editor-block-list__block-html-textarea block-editor-block-list__block-html-textarea"
-				value={ html }
-				onBlur={ this.onBlur }
-				onChange={ this.onChange }
-			/>
-		);
-	}
+	};
+
+	useEffect( () => {
+		setHtml( getBlockContent( block ) );
```

Comment:
I notice before that there was an `isEqual` check. Do you remember why? Does it make sense here?

Reply:
I think it's not necessary because the useState hook won't rerender if we end up with the same html content.

Reply:
But the `useEffect` hook will still be called? I guess it's ok to re-set the HTML?